pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -69,7 +69,70 @@ type BindMechanism interface {
 	startDHCP(vmi *v1.VirtualMachineInstance) error
 }
 
-type podNICImpl struct{}
+// Network configuration is split into two parts, or phases, each executed in a
+// different context.
+// Phase1 is run by virt-handler and heavylifts most configuration steps. It
+// also creates the tap device that will be passed by name to virt-launcher,
+// thus allowing unprivileged libvirt to consume a pre-configured device.
+// Phase2 is run by virt-launcher in the pod context and completes steps left
+// out of virt-handler. The reason to have a separate phase for virt-launcher
+// and not just have all the work done by virt-handler is because there is no
+// ready solution for DHCP server startup in virt-handler context yet. This is
+// a temporary limitation and the split is expected to go once the final gap is
+// closed.
+// Moving all configuration steps into virt-handler will also allow to
+// downgrade privileges for virt-launcher, specifically, to remove NET_ADMIN
+// capability. Future patches should address that. See:
+// https://github.com/kubevirt/kubevirt/issues/3085
+type podNIC struct {
+	vmi              *v1.VirtualMachineInstance
+	podInterfaceName string
+	iface            *v1.Interface
+	network          *v1.Network
+	handler          NetworkHandler
+}
+
+func newPodNIC(vmi *v1.VirtualMachineInstance, network *v1.Network, handler NetworkHandler) (*podNIC, error) {
+	if network.Pod == nil && network.Multus == nil {
+		return nil, fmt.Errorf("Network not implemented")
+	}
+	nic := podNIC{
+		handler:          handler,
+		vmi:              vmi,
+		network:          network,
+		podInterfaceName: composePodInterfaceName(vmi, network),
+	}
+	for i, iface := range nic.vmi.Spec.Domain.Devices.Interfaces {
+		if iface.Name == nic.network.Name {
+			nic.iface = &nic.vmi.Spec.Domain.Devices.Interfaces[i]
+		}
+	}
```

Comment:
Please export this into a function, then I think you can

```golang
     correspondingNetworkiface, err = getIfaceByNetworkName(nic.network.Name, vmi)  // or in 2 steps.
     if err != nil {
     // whatever apocalyptic error msg
     }
     return &podNIC{
          handler:          handler,
          vmi:              vmi,
          network:          network,
          podInterfaceName: composePodInterfaceName(vmi, network),
          iface:            correspondingNetworkiface,
     }, nil
```

Reply:
Good, will do it.